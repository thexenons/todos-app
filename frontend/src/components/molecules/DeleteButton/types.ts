import { ButtonProps } from "../../atoms/Button/types";

export interface DeleteButtonProps {
	disabled?: ButtonProps["disabled"];
}
