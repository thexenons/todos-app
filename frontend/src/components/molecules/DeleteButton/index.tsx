import { FC } from "react";
import { MdDeleteOutline } from "react-icons/md";

import Button from "../../atoms/Button";
import { DeleteButtonProps } from "./types";

const DeleteButton: FC<DeleteButtonProps> = ({ disabled = false, onClick }) => (
	<Button
		variant={Button.variants.contained}
		colorVariant={Button.colorVariants.error}
		disabled={disabled}
		onClick={onClick}
	>
		<Button.Icon>
			<MdDeleteOutline />
		</Button.Icon>
		Delete
	</Button>
);

export default DeleteButton;
