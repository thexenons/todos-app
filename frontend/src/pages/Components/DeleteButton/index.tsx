import { FC } from "react";

import DeleteButton from "../../../components/molecules/DeleteButton";
import * as SS from "../styled";

const DeleteButtonComponenstPage: FC = () => (
	<SS.ComponentsListWrapper>
		<h1>DeleteButton</h1>
		<SS.ComponentsRowWrapper>
			<DeleteButton />
			<DeleteButton disabled />
		</SS.ComponentsRowWrapper>
	</SS.ComponentsListWrapper>
);

export default DeleteButtonComponenstPage;
