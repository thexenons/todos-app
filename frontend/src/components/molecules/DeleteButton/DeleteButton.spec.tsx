import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import DeleteButton from ".";

describe("DeleteButton", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<DeleteButton />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
