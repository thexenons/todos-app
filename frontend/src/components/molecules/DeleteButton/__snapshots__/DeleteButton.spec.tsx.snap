// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteButton > Render 1`] = `
<div>
  <button
    class="styled__ButtonWrapper-sc-1fzdj0n-0 jhkPPx"
  >
    <span
      class="styled__ButtonIconsWrapper-sc-1fzdj0n-1 iwTLph"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </span>
    Delete
  </button>
</div>
`;
